File: frontend/src/types/post.ts
// frontend/src/types/post.ts
/**
 * Type definitions for Post entities matching the backend API.
 */

export interface Post {
  id: number;
  title: string;
  description: string;
  summary: {
    summary: string;
    keywords: string[];
  };
  file_path: string | null;
  created_at: string;
}

File: backend/app/domain/entities/post.py
# backend/app/domain/entities/post.py
from __future__ import annotations

from dataclasses import dataclass, field
from datetime import UTC, datetime
from typing import Any


@dataclass(frozen=True)
class Post:
    """Domain entity representing a post."""

    title: str
    description: str
    id: int | None = None
    summary: dict[str, Any] = field(default_factory=dict)
    file_path: str | None = None
    created_at: datetime = field(
        default_factory=lambda: datetime.now(UTC),
    )
